cg/store/crud/read.py

```diff
@@ -838,6 +822,18 @@ def get_applications(self) -> list[Application]:
             .all()
         )
 
+    def get_applications_by_ordertype(self, order_type: str) -> list[Application]:
+        """Return all possible not archived applications for an order type."""
+        non_archived_applications: Query = apply_application_filter(
+            applications=self._get_join_application_ordertype_query(),
+            filter_functions=[ApplicationFilter.IS_NOT_ARCHIVED],
+        )
+        return apply_ordertype_application_filter(
+            ordertype_applications=non_archived_applications,
+            filter_functions=[OrderTypeApplicationFilter.BY_ORDER_TYPE],
+            order_type=order_type,
+        ).all()
```

Comment:
I think we should start raising errors when something cannot be returned that is expected to have entries. In this case it would be a missing link between an order type and application. Which should be checked for as we shouldn't support an ordertype that has no application yet as the application is inherently tied to invoicing.

Reply:
fixed